python/sglang/srt/layers/moe/ep_moe/layer.py

```diff
@@ -267,18 +277,34 @@ def forward(self, hidden_states: torch.Tensor, router_logits: torch.Tensor):
             self.w13_input_scale = max_value / torch.finfo(self.fp8_dtype).max
 
         # PreReorder
-        pre_reorder_triton_kernel[(hidden_states.shape[0],)](
-            hidden_states,
-            gateup_input,
-            src2dst,
-            topk_ids,
-            self.w13_input_scale,
-            self.start_expert_id,
-            self.end_expert_id,
-            self.top_k,
-            hidden_states.shape[1],
-            BLOCK_SIZE=512,
-        )
+        if global_server_args_dict["attention_backend"] == "triton":
```

Comment:
The logic here is strange. Why check the attention backend? The MOE part and attention are independent operators.